### PR TITLE
Issue/565 extension loading

### PR DESCRIPTION
**Pull request by wouterdb**

### CODE REVIEW

File: src/inmanta/server/bootloader.py [hunk 1]
```diff
@@ -31,15 +69,77 @@ def get_server_slice(self) -> server.Server:
...
-        return AgentManager(self.restserver)
+        return AgentManager()
 
-    def get_server_slices(self) -> List[ServerSlice]:
+    def get_bootstrap_slices(self) -> List[ServerSlice]:
         return [self.get_server_slice(), self.get_agent_manager_slice()]
 
     async def start(self) -> None:
-        for mypart in self.get_server_slices():
+        for mypart in self.load_slices():
```
Comment (arnaudsjs) [inline src/inmanta/server/bootloader.py]: self.restserver.add_slice() doesn't keep track of the order in which the slices were added to the Server.

File: setup.py [hunk 2]
```diff
@@ -47,7 +49,7 @@
     },
     # Packaging
     package_dir={"": "src"},
-    packages=find_packages("src"),
+    packages=find_packages("src") + namespace_packages,
```
Comment (arnaudsjs) [inline setup.py]: Add comment to explain why namespace_packages is required.

File: setup.py [hunk 3]
```diff
@@ -47,7 +50,7 @@
...
```
Comment (bartv) [inline setup.py]: Do we need to add namespace_package and use find_namespace_packages?

https://setuptools.readthedocs.io/en/latest/setuptools.html#namespace-packages
https://setuptools.readthedocs.io/en/latest/setuptools.html#find-namespace-packages

https://setuptools.readthedocs.io/en/latest/pkg_resources.html?#namespace-package-support

File: src/inmanta/server/protocol.py [hunk 4]
```diff
@@ -113,11 +151,17 @@ def get_id(self) -> str:
         await self.connect_database()
 
         for slice in self.get_slices().values():
```
Comment (bartv) [inline src/inmanta/server/protocol.py]: slice is actually a built-in function

File: src/inmanta/server/server.py [hunk 5]
```diff
@@ -120,7 +121,11 @@ def __init__(self, agent_no_log: bool = False) -> None:
         self._increment_cache_locks: Dict[uuid.UUID, locks.Lock] = defaultdict(lambda: locks.Lock())
         self._influx_db_reporter: Optional[InfluxReporter] = None
 
+    def get_dependencies(self) -> List[str]:
+        return [SLICE_SESSION_MANAGER]
+
     async def prestart(self, server: protocol.Server) -> None:
+        self._server_storage: Dict[str, str] = self.check_storage()
         self.agentmanager: "AgentManager" = server.get_slice("agentmanager")
```
Comment (bartv) [inline src/inmanta/server/server.py]: agentmanager vs session manager.

File: setup.py (hunk 3 above)
Comment (wouterdb) [reply setup.py]: the find_namespace_packages is a good suggestion


 pkg_resources.declare_namespace approach is no longer recommended as per https://packaging.python.org/guides/packaging-namespace-packages/#pkg-resources-style-namespace-packages

**Pull request merged**